src/main/java/fi/aalto/cs/intellij/util/StringSplitter.java

```diff
@@ -0,0 +1,71 @@
+package fi.aalto.cs.intellij.util;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A class that (hopefully) helps splitting strings.
+ * Usage:
+ * <pre>
+ * {@code
+ * StringSplitter s = new StringSplitter("hello-123-world", "-");
+ * s.readNext();     // returns "hello"
+ * s.readNextInt();  // returns 123
+ * s.readNext();     // returns "world"
+ * }
+ * </pre>
+ */
+public class StringSplitter {
```

Comment:
Sonar wasn't happy with regexes so I created this small helper class to parse strings. :(

Reply:
can something like this be used https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-
there are regexes under the hood as well. but as it's Java's own API, then sonar should not care:
`  
  @Test
  public void testStringSplit() {
    String string = "John,,Paul,Ringo Starr,George,";
    String[] stringsArray = string.split(",");

    Assert.assertEquals("John", stringsArray[0]);
    Assert.assertEquals("", stringsArray[1]);
    Assert.assertEquals("Paul", stringsArray[2]);
    Assert.assertEquals("Ringo Starr", stringsArray[3]);
    Assert.assertEquals("George", stringsArray[4]);

//    this fails, as the last trailing is being cut off
//    Assert.assertEquals("", stringsArray[5]);
//
  }
`
 Or the behavior of preserving the last empty "trailing" string is critical?

Reply:
I considered that but I was afraid that Sonar would still keep nagging because the separator has to be given as an oh-so-dangerous regex. I can check what happens. Would be nice if we could get rid of this class.
Other chance could be introduce something like Apache Commons library to our project, and use utility functions provided by that.

Reply:
You were right! Sonar doesn't consider `String.split(String regex)` as a "security hotspot" (as it did with a general use of regex). So I removed this class. 👍